opentelemetry-user-events-logs/src/logs/processor.rs

```diff
@@ -93,7 +94,18 @@ impl<'a> ProcessorBuilder<'a> {
     /// For example the following will capture level 2 (Error) and 3(Warning) events:
     /// perf record -e user_events:myprovider_L2K1,user_events:myprovider_L3K1
     pub(crate) fn new(provider_name: &'a str) -> Self {
-        Self { provider_name }
+        Self {
+            provider_name,
+            resource_attribute_keys: vec![],
+        }
+    }
+
+    /// Sets the resource attributes for the processor
+    /// This is the list of attribute keys, which, if present in Resource attributes,
+    /// will be exported with each log.
+    pub fn with_resource_attributes<S: Into<String>>(mut self, attributes: Vec<S>) -> Self {
```

Comment:
We could make the API more ergonomic for the user by accepting an iterator instead of a vec. You could also use `Cow<'static,str>` if we want to avoid allocation for static strings.

```suggestion
    pub fn with_resource_attributes<I, S>(mut self, attributes: I) -> Self
    where
        I: IntoIterator<Item = S>,
        S: Into<Cow<'static, str>>,
```